src/views/catalog/templatescatalog/components/TemplatesCatalogDrawer/TemplatesCatalogDrawer.tsx

```diff
@@ -0,0 +1,74 @@
+import * as React from 'react';
+import { useTranslation } from 'react-i18next';
+
+import { V1Template } from '@kubevirt-ui/kubevirt-api/console';
+import { CatalogItemHeader } from '@patternfly/react-catalog-view-extension';
+import { Modal, Split, SplitItem } from '@patternfly/react-core';
+
+import {
+  getTemplateName,
+  getTemplateProviderName,
+  getTemplateSupportLevel,
+} from '../../utils/helpers';
+import { getTemplateOSIcon } from '../../utils/os-icons';
+
+import { TemplatesCatalogDrawerFooter } from './TemplatesCatalogDrawerFooter';
+import { TemplatesCatalogDrawerPanel } from './TemplatesCatalogDrawerPanel';
+
+import './TemplateCatalogDrawer.scss';
+
+export const TemplatesCatalogDrawer: React.FC<{
+  namespace: string;
+  template: V1Template | undefined;
+  isOpen: boolean;
+  onClose: () => void;
+}> = React.memo(({ namespace, template, isOpen, onClose }) => {
+  const { t } = useTranslation('plugin__kubevirt-plugin');
+  const provider = getTemplateProviderName(template);
+  const support = getTemplateSupportLevel(template);
+
+  return (
+    <Modal
+      aria-label="template drawer"
+      className="ocs-modal co-catalog-page__overlay co-catalog-page__overlay--right template-catalog-drawer"
+      isOpen={isOpen}
+      onClose={onClose}
+      header={
+        template && (
+          <CatalogItemHeader
+            className="co-catalog-page__overlay-header"
+            title={getTemplateName(template)}
+            vendor={
+              <Split hasGutter>
+                <SplitItem>
+                  <strong>{t('Provided by: ')}</strong> {provider}
+                </SplitItem>{' '}
+                <SplitItem> | </SplitItem>
+                <SplitItem>
+                  <strong>{t('Support level: ')}</strong> {support || 'N/A'}
+                </SplitItem>
+                <SplitItem> | </SplitItem>
+                <SplitItem>
+                  <strong>{t('Supports quick create VM')}</strong>
+                </SplitItem>
+              </Split>
+            }
+            iconImg={getTemplateOSIcon(template)}
+          />
+        )
+      }
+      footer={
+        template && (
+          <TemplatesCatalogDrawerFooter
+            namespace={namespace}
+            template={template}
+            onCreate={(tmp) => console.log(tmp)}
+            onCancel={onClose}
+          />
+        )
+      }
+    >
+      {template && <TemplatesCatalogDrawerPanel template={template} />}
+    </Modal>
+  );
+});
```

Comment:
i can see in this file 3 times template && , consider something like const obj = template ? {footter: React.FC, ...} {footer: null, ...}